src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -64,6 +56,18 @@
       (assoc :file file)
       (assoc :line line)))
 
+(schema/defn query-param :- Rule
+  "Add a query parameter matching value(s) to a rule. New values will be
+   appended to existing values.
+
+   The query parameters are in a map under the `:match-request` section of the
+   rule. Keys in the map are strings corresponding to the query parameters to
+   match, and the values are sequences of strings of acceptable values."
+  [rule :- Rule
+   param :- schema/Str
+   value :- (schema/either schema/Str [schema/Str])]
+  (update-in rule [:query-params param] into (flatten [value])))
```

Comment:
Ugh this line doesn't behave the way we want if we try to make them sets here.  I thought this was just a matter of wrapping a `set` call around `flatten`, but it's not.

I'll continue playing with it